demos/CMakeLists.txt

```diff
@@ -7,6 +7,34 @@ if( NOT DEFINED PLATFORM_NAME )
     endif()
 endif()
 
+# This function will add a demo target to the list of default targets
+# if all AWS Credentials are defined.
+function(check_aws_credentials demo_name)
+    if(AWS_IOT_ENDPOINT AND CLIENT_CERT_PATH AND CLIENT_PRIVATE_KEY_PATH)
+        target_compile_definitions(
+            ${demo_name} PRIVATE
+                AWS_IOT_ENDPOINT="${AWS_IOT_ENDPOINT}"
+                CLIENT_CERT_PATH="${CLIENT_CERT_PATH}"
+                CLIENT_PRIVATE_KEY_PATH="${CLIENT_PRIVATE_KEY_PATH}"
+        )
+        set(CMAKE_REQUIRED_DEFINITIONS -DAWS_IOT_ENDPOINT -DCLIENT_CERT_PATH -DCLIENT_PRIVATE_KEY_PATH)
+    endif()
```

Comment:
Question: Is this so that parameters passed from the command line will be counted as well? I think the current code ignores command line parameters when determining build targets so that's a good change to have.

Reply:
Yup, this will allow the user to set the credentials without having to modify the demo config header.

Reply:
One thing I learned is that you can automatically set macros by just passing a CMake command line parameter: `-DSAMPLE_MACRO=1`

However, it doesn't work for strings unless you set it manually with `target_compile_definitions`.